pkg/http/handler/timeout.go

```diff
@@ -173,3 +175,23 @@ func (tw *timeoutWriter) timeoutAndWriteError(msg string) bool {
 
 	return false
 }
+
+var timerPool sync.Pool
+
+func getTimer(timeout time.Duration) *time.Timer {
+	if v := timerPool.Get(); v != nil {
+		t := v.(*time.Timer)
+		t.Reset(timeout)
+		return t
+	}
+	return time.NewTimer(timeout)
+}
+
+func putTimer(t *time.Timer, alreadyDrained bool) {
+	if !t.Stop() && !alreadyDrained {
```

Comment:
I think I understand now (just commenting here for future readers in case it's helpful, because the semantics of Stop and the race this is handling here were not immediately clear to me) that what's really happening here is:

~~~~
   timerAlreadyFired := !t.Stop()
   if timerAlreadyFired && !alreadyDrained {
     // We Stopped, but the timer had already fired and we didn't read from the channel yet;
     // this happens if both t.C and done have a value and the select decides to go via the done case
     // before the t.C case consumes the message.
     <- t.C
  }
~~~~

don't know if we need all this verbosity in the actual comment here, but maybe a few more words about what's happening are worth it.

Reply:
Added a more verbose comment, please check!

Reply:
Yeah, perfect 👍